exampleSubscription/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/gorilla/websocket"
+)
+
+type MyDialer struct {
+	*websocket.Dialer
+}
+
+func (md *MyDialer) DialContext(ctx context.Context, urlStr string, requestHeader http.Header) (graphql.WSConn, *http.Response, error) {
+	conn, resp, err := md.Dialer.DialContext(ctx, urlStr, requestHeader)
+	return graphql.WSConn(conn), resp, err
+}
+
+func main() {
+	var err error
+	defer func() {
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}()
+
+	key := os.Getenv("GITHUB_TOKEN")
+	if key == "" {
+		err = fmt.Errorf("must set GITHUB_TOKEN=<github token>")
+		return
+	}
+
+	dialer := websocket.DefaultDialer
+	headers := http.Header{}
+	headers.Add("Authorization", "bearer "+key)
+
+	graphqlClient := graphql.NewClientUsingWebSocket(
+		"wss://api.github.com/graphql",
+		&MyDialer{Dialer: dialer},
+		headers,
```

Comment:
Wait, does the real GitHub API even have subscriptions? If not, I don't think this will work, right? We could either include a gqlgen server with the example, or just skip the full example in the repo entirely and provide the relevant snippets in the documentation. (Eventually maybe we should have many examples each with their own gqlgen server.)

Reply:
I added the example thinking we could add a subscription on the server side.
But for now I will just replace it with documentation.